tests/network_rules_update_test.go

```diff
@@ -0,0 +1,204 @@
+package tests
+
+import (
+	"encoding/json"
+	"github.com/0xsoniclabs/sonic/evmcore"
+	"github.com/0xsoniclabs/sonic/gossip/contract/driverauth100"
+	"github.com/0xsoniclabs/sonic/opera/contracts/driverauth"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/stretchr/testify/require"
+	"math/big"
+	"testing"
+)
+
+func TestNetworkRule_Update_RulesChangeDuringEpochHasNoEffect(t *testing.T) {
+	require := require.New(t)
+	net := StartIntegrationTestNetWithFakeGenesis(t)
+	defer net.Stop()
+
+	client, err := net.GetClient()
+	require.NoError(err)
+	defer client.Close()
+
+	type rulesType struct {
+		Economy struct {
+			MinBaseFee *big.Int
+		}
+	}
+
+	var originalRules rulesType
+	err = client.Client().Call(&originalRules, "eth_getRules", "latest")
+	require.NoError(err)
+	require.NotEqual(0, originalRules.Economy.MinBaseFee.Int64(), "MinBaseFee should be filled")
+
+	newMinBaseFee := 1e3 * originalRules.Economy.MinBaseFee.Int64()
+	updateRequest := rulesType{}
+	updateRequest.Economy.MinBaseFee = new(big.Int).SetInt64(newMinBaseFee)
+
+	// Update network rules
+	updateNetworkRules(t, net, updateRequest)
+
+	// Network rule should not change - it must be an epoch bound
+	var updatedRules rulesType
+	err = client.Client().Call(&updatedRules, "eth_getRules", "latest")
+	require.NoError(err)
+
+	require.Equal(originalRules.Economy.MinBaseFee, updatedRules.Economy.MinBaseFee,
+		"Network rules should not change - it must be an epoch bound")
+
+	var blockBefore evmcore.EvmBlockJson
+	err = client.Client().Call(&blockBefore, "eth_getBlockByNumber", "latest", false)
+	require.NoError(err)
```

Comment:
For most common operations, the client library has a few ready-to-use features such that you do not have to know the explicit RPC methods. For this one, for instance, you can use
```suggestion
	blockBefore, err := client.BlockByNumber(t.Context(), nil)
	require.NoError(err)
```